security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -649,11 +649,16 @@ func (sc *SecretManagerClient) registerSecret(item security.SecretItem) {
 	sc.cache.SetWorkload(&item)
 	resourceLog(item.ResourceName).Debugf("scheduled certificate for rotation in %v", delay)
 	sc.queue.PushDelayed(func() error {
-		resourceLog(item.ResourceName).Debugf("rotating certificate")
-		// Clear the cache so the next call generates a fresh certificate
-		sc.cache.SetWorkload(nil)
-
-		sc.OnSecretUpdate(item.ResourceName)
+		// In case `UpdateConfigTrustBundle` called, it will resign workload cert.
+		// Check if this is a stale scheduled rotating task.
+		if cached := sc.cache.GetWorkload(); cached != nil {
+			if cached.CreatedTime == item.CreatedTime {
+				resourceLog(item.ResourceName).Debugf("rotating certificate")
+				// Clear the cache so the next call generates a fresh certificate
+				sc.cache.SetWorkload(nil)
```

Comment:
This is outside of this PR, but we may not want to do `delete` + `trigger fetch`.

if the fetch fails, we will have no cert and an issue. Better to keep the old one until we fetch the new one

Probably doesn't matter though since Envoy isn't continually requesting it...

Reply:
If we donot trigger push, when will envoy automatically fetch fresh new certificate

Reply:
I mean trigger push, but do not remove the SetWorkload. E.g in getCachedSecret() we regen if the cert is near expiry instead of deleting it entirely.

Probably not a big deal in practice..